pkg/cmd/repo/garden/garden.go

```diff
@@ -216,6 +219,22 @@ func gardenRun(opts *GardenOptions) error {
 	_ = exec.Command("stty", sttyFileArg, "/dev/tty", "cbreak", "min", "1").Run()
 	_ = exec.Command("stty", sttyFileArg, "/dev/tty", "-echo").Run()
 
+	walkAway := func() {
```

Comment:
IMO this would be good to extract into normal function. Is there a reason it must be defined inline like this?

Reply:
@samcoe Good question! This is an example of using a closure like in JavaScript. The references to `opts`, `out`, and `oldTTYSettings` are provided to us by the closure.

If the `walkAway()` function was defined top-level, we could invoke it, but we would have to pass all the references as arguments. It wouldn't be a huge deal, but in this case, a closure is convenient and we're not planning to call `walkAway()` from anywhere else, so I'm fine with this approach.